lib/dialyzer/src/dialyzer_contracts.erl

```diff
@@ -933,28 +936,34 @@ t_from_forms_without_remote(_Forms, _MFA, _File, _RecDict) ->
   %% Lots of forms
   unsupported.
 
-blame_remote(ContractSig, NoRemoteContractSig, Sig) ->
+blame_remote(ContractSig, NoRemoteContractSig, Sig, Opaques) ->
   CArgs  = erl_types:t_fun_args(ContractSig),
   CRange = erl_types:t_fun_range(ContractSig),
   NRArgs = erl_types:t_fun_args(NoRemoteContractSig),
   NRRange = erl_types:t_fun_range(NoRemoteContractSig),
   SArgs = erl_types:t_fun_args(Sig),
   SRange = erl_types:t_fun_range(Sig),
-  blame_remote_list([CRange|CArgs], [NRRange|NRArgs], [SRange|SArgs]).
+  blame_remote_list([CRange|CArgs], [NRRange|NRArgs], [SRange|SArgs], Opaques).
 
-blame_remote_list([], [], []) ->
+blame_remote_list([], [], [], _Opaques) ->
   true;
-blame_remote_list([CArg|CArgs], [NRArg|NRArgs], [SArg|SArgs]) ->
+blame_remote_list([CArg|CArgs], [NRArg|NRArgs], [SArg|SArgs], Opaques) ->
   case erl_types:t_is_equal(CArg, NRArg) of
     true ->
       case not erl_types:t_is_equal(CArg, SArg) of
-	true  -> false;
-	false -> blame_remote_list(CArgs, NRArgs, SArgs)
+        true  -> false;
+        false -> blame_remote_list(CArgs, NRArgs, SArgs, Opaques)
       end;
     false ->
-      case erl_types:t_is_subtype(SArg, NRArg)
-	andalso not erl_types:t_is_subtype(NRArg, SArg) of
-	true  -> false;
-	false -> blame_remote_list(CArgs, NRArgs, SArgs)
+      case is_subtype(SArg, NRArg, Opaques)
+        andalso not is_subtype(NRArg, SArg, Opaques) of
+        true  -> false;
+        false -> blame_remote_list(CArgs, NRArgs, SArgs, Opaques)
       end
   end.
+
+%% As erl_types:t_is_subtype/2 but without looking into opaque types that
+%% aren't known to us.
+is_subtype(T1, T2, Opaques) ->
+    Inf = erl_types:t_inf(T1, T2, Opaques),
+    erl_types:t_is_equal(T1, Inf).
```

Comment:
Nitpick: dialyzer source files should be indented with 2 spaces, not 4. (There is an `-*- erlang-indent-level: 2 -*-` comment at the top of all files.)

Reply:
Thanks, fixed locally.